source/extensions/filters/http/tap/tap_filter.cc

```diff
@@ -69,6 +69,13 @@ Http::FilterTrailersStatus Filter::encodeTrailers(Http::ResponseTrailerMap& trai
   return Http::FilterTrailersStatus::Continue;
 }
 
+void Filter::setDownstreamConnectionAddress(
+    const Envoy::Network::ConnectionInfoProvider& connection_info_provider) {
+  if (tapper_ != nullptr) {
+    tapper_->setDownstreamConnectionAddress(connection_info_provider);
+  }
+}
+
```

Comment:
You shouldn't need all these from StreamFilterBase to here. The downstream connection is available via StreamFilterCallbacks::connection, or you can get them in the log method below from StreamInfo.

Reply:
@lizan Thanks, you're right. I have changed code to get the connection from StreamFilterCallbacks::connection, please help to review this again.